### PR TITLE
add validation test for clusters

### PR DESCRIPTION
**Pull request by sonchang**

### CODE REVIEW

File: tests/validation/cattlevalidationtest/core/test_cluster.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            networkIds=[managed_network.id],
+            imageUuid='docker:ubuntu',
+            requestedHostId=cluster.id)
+        wait_for_condition(
+            client, test_container,
+            lambda x: x.state == 'running',
+            lambda x: 'State is: ' + x.state)
+
+        test_container_found = None
+        for instance in cluster.instances():
```
Comment (alena1108) [inline tests/validation/cattlevalidationtest/core/test_cluster.py]: Son, can we also check that the network was set for the instances? As those instances go through the separate allocator. Just check that the vlan/subnet was set, and the instance got an ip from the managed network

File: tests/validation/cattlevalidationtest/core/common_fixtures.py [hunk 2]
```diff
@@ -167,3 +168,23 @@ def ping_link(src, link_name, var=None, value=None):
             time.sleep(1)
 
     assert from_link == value
+
+
+@pytest.fixture
```
Comment (alena1108) [inline tests/validation/cattlevalidationtest/core/common_fixtures.py]: We could benefit from this function in our integration tests too! :) can you add it there once this PR is merged?

File: tests/validation/cattlevalidationtest/core/test_cluster.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        socat_container1 = client.create_container(
+            name=test_name + '-socat',
+            networkIds=[managed_network.id],
+            imageUuid=socat_test_image,
+            ports='2375:2375/tcp',
+            stdinOpen=False,
+            tty=False,
+            dataVolumes='/var/run/docker.sock:/var/run/docker.sock',
+            requestedHostId=hosts[0].id)
+        wait_for_condition(
```
Comment (alena1108) [inline tests/validation/cattlevalidationtest/core/test_cluster.py]: We should also check "system_container" field of the swarm instances

File: tests/validation/cattlevalidationtest/core/test_cluster.py (hunk 1 above)
Comment (alena1108) [inline tests/validation/cattlevalidationtest/core/test_cluster.py]: Shouldn't we reload the cluster before retrieving cluster.instances()? Looks like we reference it right after the instance is deployed in the cluster, w/o reloading it.

Comment (sonchang) [inline tests/validation/cattlevalidationtest/core/test_cluster.py]: Sorry.  Accidentally closed the PR.
I updated the test to try to make more checks regarding the network.  But I wasn't able to find the vnet or subnet in the API.
cluster.instances() I think makes an API call so it doesn't look like cluster itself needs to be updated unless I'm checking fields in that specific resource.

File: tests/validation/cattlevalidationtest/core/common_fixtures.py (hunk 2 above)
Comment (sonchang) [inline tests/validation/cattlevalidationtest/core/common_fixtures.py]: I think I have a form of this already in integration tests.  :-)

File: tests/validation/cattlevalidationtest/core/test_cluster.py (hunk 3 above)
Comment (sonchang) [inline tests/validation/cattlevalidationtest/core/test_cluster.py]: Added!  Thanks!

**Pull request closed without merging**